components/mpas-ocean/src/mode_init/mpas_ocn_init_vertical_grids.F

```diff
@@ -1497,6 +1498,9 @@ subroutine ocn_init_vertical_grid_with_max_rx1(domain, iErr)
 
              if (maxLevelCell(iCell) > maxLevelNeighbors) then
                 maxLevelCell(iCell) = maxLevelNeighbors
+                if (smoothingMask(iCell) == 0) then
+                   bottomDepth(iCell) = refBottomDepth(maxLevelNeighbors)
+                end if
              end if
           end do
```

Comment:
This fix makes this treatment within the haney-number coordinate (but outside the "smoothing mask" that defines the region where the haney-number coordinate is actually applied) consistent with the same code in the z-star coordinate used without ice shelf cavities, see:
https://github.com/E3SM-Project/E3SM/blob/1e2c8e15c427ea2e7535ddad35b5a97a1cb1dd55/components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F#L831-L834